core/src/test/java/com/linecorp/armeria/client/endpoint/healthcheck/PartialHealthCheckStrategyTest.java

```diff
@@ -0,0 +1,279 @@
+            // At this time, even an unhealthy candidate delivered by the function may be randomly chosen again.
+            if (actCandidates.contains(unhealthyEndpoint)) {
+                assertThat(actRes).isFalse();
+            } else {
+                assertThat(actRes).isTrue();
+            }
+
+            assertThat(actCandidates).hasSize(5);
+            actCandidates.forEach(
+                    actCandidate -> assertThat(candidatesForMaxCount.contains(actCandidate)).isTrue());
+        }
+    }
+
+    @Test
+    void updateHealthWhenMaxRatioMode() {
+        List<Endpoint> actCandidates = maxRatioStrategy.getCandidates();
+        final Endpoint candidates = actCandidates.get(0);
+
+        assertThat(actCandidates).hasSize(9);
+
+        boolean actUpdateRes = maxRatioStrategy.updateHealth(candidates, UNHEALTHY);
+        actCandidates = maxRatioStrategy.getCandidates();
+
+        assertThat(actUpdateRes).isTrue();
+        assertThat(actCandidates).hasSize(9);
+        assertThat(actCandidates.contains(candidates)).isFalse();
+
+        actUpdateRes = maxRatioStrategy.updateHealth(candidates, HEALTHY);
+        actCandidates = maxRatioStrategy.getCandidates();
+
+        assertThat(actUpdateRes).isFalse();
+        assertThat(actCandidates).hasSize(9);
+        assertThat(actCandidates.contains(candidates)).isFalse();
+
+        actUpdateRes = maxRatioStrategy.updateHealth(actCandidates.get(0), UNHEALTHY);
+        actCandidates = maxRatioStrategy.getCandidates();
+
+        assertThat(actUpdateRes).isTrue();
+        assertThat(actCandidates).hasSize(9);
+        assertThat(actCandidates.contains(candidates)).isTrue();
+    }
+
+    @Test
+    void updateHealthWhenMaxValueMode() {
+        List<Endpoint> actCandidates = maxCountStrategy.getCandidates();
+        final Endpoint candidates = actCandidates.get(0);
+
+        assertThat(actCandidates).hasSize(5);
+
+        boolean actUpdateRes = maxCountStrategy.updateHealth(candidates, UNHEALTHY);
+        actCandidates = maxCountStrategy.getCandidates();
+
+        assertThat(actUpdateRes).isTrue();
+        assertThat(actCandidates).hasSize(5);
+        assertThat(actCandidates.contains(candidates)).isFalse();
+
+        actUpdateRes = maxCountStrategy.updateHealth(candidates, HEALTHY);
+        actCandidates = maxCountStrategy.getCandidates();
+
+        assertThat(actUpdateRes).isFalse();
+        assertThat(actCandidates).hasSize(5);
+        assertThat(actCandidates.contains(candidates)).isFalse();
+
+        actUpdateRes = maxCountStrategy.updateHealth(actCandidates.get(0), UNHEALTHY);
+        actCandidates = maxCountStrategy.getCandidates();
+
+        assertThat(actUpdateRes).isTrue();
+        assertThat(actCandidates).hasSize(5);
+        assertThat(actCandidates.contains(candidates)).isTrue();
+    }
+
+    @Test
+    void updateHealthByDisappearedCandidate() {
+        final Endpoint disappearedCandidate = Endpoint.ofGroup("disappeared");
```

Comment:
`Endpoint.of("disappeared")` ? (no group)

Reply:
fixed